app/assets/javascripts/interactive-manager.coffee

```diff
@@ -0,0 +1,34 @@
+# One place for all interactives
+
+class InteractiveManager
+  #IFRAMES_SEL = 'iframe.interactive'
+
+  @_instance = null
+
+  # PUBLIC
+  # Only class methods are designed to be public.
+  @register: (iframeEl) ->
+    @_getInstance()._register(iframeEl)
+
+  @_getInstance: ->
+    @_instance = new InteractiveManager() unless @_instance
+    @_instance
+
+  _register: ($interactive) ->
+    $iframe = $interactive.find('iframe')
+    $data = $interactive.find('.interactive_data_div')
+    $delete_button = $interactive.find('.delete_interactive_data')
+    $click_to_play = $interactive.find('.click_to_play.shown')
+
+    if $click_to_play.length == 0
+      if $interactive.hasClass('savable') > 0
+        new IFrameSaver($iframe, $data, $delete_button)
+
+      if gon.globalInteractiveState?
+        window.globalIframeSaver = new GlobalIframeSaver gon.globalInteractiveState, $iframe
```

Comment:
It's not going to work.

`GlobalIframeSaver` should have only once instance that manages all the interactives. Perhaps it should have been implemented as singleton too.

You should use existing instance and just call `addNewPhone()` (the only public method there, exactly due to his reason). Probably it would be better to change `addNewPhone` into `addNewInteractive` and get phone within this method, so we don't have to do it here.